### PR TITLE
Drag and drop 864

### PR DESCRIPTION
**Pull request by miqur**
@squallstar 
screenshot1
screenshot2
screenshot3
screenshot4
screenshot5

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -518,9 +519,33 @@ function toggleAll(el) {
...
+    var idType = '';
+    var dataType = '';
 
     for (var i = 0; i < navStack.length; i++) {
-      breadcrumbsPath += '<span class="bread-link"><a href="#" data-breadcrumb="' + i + '">' + navStack[i].name + '</a></span>';
+      switch (navStack[i].type) {
+        case 'organizationId':
+          type = 'organization';
+          idType = 'data-org-id=';
+          dataType = 'data-type=';
```
Comment (squallstar) [inline js/interface.js]: why do these contain the `=` sign? surely that should be part of the initial template

File: js/interface.js [hunk 2]
```diff
@@ -903,6 +928,112 @@ function showNothingFoundAlert(isShow) {
   }
 }
 
+function showAlertWithButtonGoToFolder(item, area) {
+  var goToButton = $('#alert-btn-action').attr('data-type', area.type);
+
+  if (area.type == 'organization') {
```
Comment (squallstar) [inline js/interface.js]: `===`, please check the code guidelines

File: js/interface.js [hunk 3]
```diff
@@ -903,6 +928,112 @@ function showNothingFoundAlert(isShow) {
...
+    goToButton.attr('data-id', area.appId);
+  } else {
+    goToButton.attr('data-id', area.id);
+  }
+
+  $('.alert-action').addClass('active');
+  $('.alert-message').text(item.length + ' item(s) moved');
+  setTimeout(function() {
+    setAlertVisibilityWhenMovingItems(false);
+  }, $goToFolderAlertTimeout);
```
Comment (squallstar) [inline js/interface.js]: the `$` prefix is used for jquery variables, this should be an integer instead

File: js/interface.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ var $searchType = $('.search-type');
 var $searchTerm = $('.search-term');
 var $fileTable = $('.file-table');
 var $pagination = $('.pagination');
+var $goToFolderAlertTimeout = 5000;
```
Comment (squallstar) [inline js/interface.js]: only use the `$` prefix for jquery elements

File: js/interface.js [hunk 5]
```diff
@@ -903,6 +928,112 @@ function showNothingFoundAlert(isShow) {
...
+  } else if (dropZone.fileType === 'folder') {
+    movedPlace[parent] = dropZone.id;
+  } else {
+    return false;
+  }
+  return movedPlace;
+}
+
+function moveItems(folderType, id, dropArea, element, items) {
+  if (folderType === 'folder') {
```
Comment (squallstar) [inline js/interface.js]: no error caching for these API requests?

File: js/interface.js [hunk 6]
```diff
@@ -903,6 +928,112 @@ function showNothingFoundAlert(isShow) {
...
+  }
+}
+
+// Go to folder where dropped items
+$('#alert-btn-action').on('click', function() {
+  var $button = $(this);
+  var dataType = $button.attr('data-type');
+  var dataIdAttribute = $button.attr('data-id');
+
+  if (dataType == 'organization') {
```
Comment (squallstar) [inline js/interface.js]: `setAlertVisibilityWhenMovingItems(false);` is repeated three timed in the following lines

File: js/interface.js [hunk 7]
```diff
@@ -931,6 +1062,10 @@ $dropZone.on('dragleave', function(e) {
 
 $('html').on('dragenter', function(e) {
   e.preventDefault();
+  if(e.originalEvent.dataTransfer.files.length === 0) {
```
Comment (squallstar) [inline js/interface.js]: space after `if`..

File: js/interface.js [hunk 8]
```diff
@@ -1288,7 +1423,61 @@ $('.file-manager-wrapper')
...
+  })
+  .on('drop', '.drop-area', function(e) {
+    e.preventDefault();
+    var area = $(this);
+    var dropArea = area.data();
+    var items = $('.file-row.active');
+    var itemType = JSON.parse(e.originalEvent.dataTransfer.getData('text'));
+    var $element = $('.file-row[data-id=' + itemType.id + ']');
+    var checkDraggedItems = checkDraggedFileIfSelected(itemType, items);
+    var alertConfirmationMovingItem = confirm('Are you sure you want to move item(s)?');
```
Comment (squallstar) [inline js/interface.js]: `Are you sure you want to move these item(s)?`

File: js/interface.js [hunk 9]
```diff
@@ -1288,7 +1423,61 @@ $('.file-manager-wrapper')
...
+        $('.alert-message').text('Moving ' + items.length + ' items...');
+        $(items).each(function() {
+          var $element = $(this);
+          var folderType = $element.attr('data-file-type');
+          var itemId = $element.attr('data-id');
+          setOpacityWhenMovingItems($element);
+          moveItems(folderType, itemId, dropArea, $element, items);
+        });
+      } else {
+        $('.alert-message').text('Moving ' + $element.length + ' items...');
```
Comment (squallstar) [inline js/interface.js]: item(s)

File: js/interface.js [hunk 10]
```diff
@@ -903,6 +928,112 @@ function showNothingFoundAlert(isShow) {
   }
 }
 
+function showAlertWithButtonGoToFolder(item, area) {
```
Comment (squallstar) [inline js/interface.js]: `showGoToFolderAlert`

File: js/interface.js (hunk 1 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 2 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 3 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 4 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 5 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 6 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 7 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 8 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 9 above)
Comment (miqur) [reply js/interface.js]: fixed

File: js/interface.js (hunk 10 above)
Comment (miqur) [reply js/interface.js]: done

**Pull request merged**